runtime/src/accounts_db.rs

```diff
@@ -1874,6 +1874,27 @@ pub(crate) struct ShrinkAncientStats {
     pub(crate) ancient_scanned: AtomicU64,
 }
 
+#[derive(Debug, Default)]
+pub(crate) struct ShrinkStatsSub {
+    store_accounts_timing: StoreAccountsTiming,
+    rewrite_elapsed_us: u64,
+    create_and_insert_store_elapsed_us: u64,
+}
+
+impl ShrinkStatsSub {
+    #[allow(dead_code)]
+    pub(crate) fn accumulate(&mut self, other: &Self) {
```

Comment:
What's the intent for this in the future? I'm not seeing a great place that it'd be used right now, which is influencing whether I think this change is very useful 😄

Reply:
Multiple ancient append vecs will soon be built/packed/combined from multiple source slots & append vecs in 1 pass across many slots with several wrinkles. These stats will be accumulated together across multiple scans, writes, creates, etc.

Reply:
and the code to do this is a whole new set of code that will exist side by side for a while. Trying to make that change as easy to understand as possible. So, this stats refactoring makes more shared code and code that stays the same when the new code is put in. New code is working. I'm writing tests for all the pieces now.